packages/frontend/src/utils/axios.ts

```diff
@@ -9,10 +10,45 @@ const isJsonBlob = (data): data is Blob =>
  *
  * @param err
  */
-export const handleErrors = (
-  err: AxiosError,
+export const handleErrors = async (
+  err: AxiosError<{
+    code: number;
+    message: string;
+    statusCode: number;
+  }>,
   handleErrorsAutomatically = true
-): AxiosError => {
+): Promise<
+  AxiosError<{
+    code: number;
+    message: string;
+    statusCode: number;
+  }>
+> => {
+  let refreshToken;
+  const recoilPersist = localStorage.getItem('recoil-persist');
+  if (recoilPersist) {
+    refreshToken = JSON.parse(recoilPersist)?.ActiveTokenSet?.refreshToken;
+  }
+
+  if (
+    refreshToken &&
+    err?.response?.status === 401 &&
+
+    // 1092, 1107 are the error codes for invalid jwt token that defined in backend
+    (err?.response?.data?.code === 1092 || err?.response?.data?.code === 1107)
+  ) {
+    // delete the old token from localStorage to prevent infinite loop
+    delete recoilPersist!['ActiveTokenSet'];
+    localStorage.setItem(JSON.stringify(recoilPersist), 'recoil-persist');
```

Comment:
@kristoferlund 
It seems we just can use `recoil` inside components so I had to manually put/get to local storage